nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,58 @@
+const express = require("express");
+const router = express.Router();
+const meals = require("./../data/meals.json");
+
+/* Respond with the json for all the meals */
+/* router.get("/", async (req, res) => {
+  try {
+    res.json(meals);
+  } catch (error) {
+    throw error;
+  }
+});  */
+
+/* Respond with the json for the meal with the corresponding id */
+router.get("/:id", async (req, res) => {
+  try {
+    const result = getMeal(parseInt(req.params.id));
+    Object.keys(result).length === 0
+      ? res.status(404).json({ error: `${req.params.id} not found` })
+      : res.status(200).json(result);
+  } catch (error) {
+    throw new Error();
+  }
+});
+
+const getMeal = (id) => {
+  return meals.filter((meal) => meal.id === id);
+};
+
+router.get("/", async (req, res) => {
+  try {
+    const { maxPrice, title, createdAfter, limit } = req.query;
+    let filteredMeals = meals;
+
+    if (Object.keys(req.query).length === 0) {
+    } else {
```

Comment:
Generally try to avoid empty if clauses, instead you could just do if (Object.keys(req.query).length != 0)

Reply:
Changed it to if (Object.keys(req.query).length != 0) as suggested.